amazon-kinesis-client/src/test/java/software/amazon/kinesis/retrieval/fanout/FanOutConfigTest.java

```diff
@@ -68,11 +77,24 @@ public void testRegisterCalledWhenConsumerArnUnset() throws Exception {
         ShardInfo shardInfo = mock(ShardInfo.class);
 //        doReturn(Optional.of(StreamIdentifier.singleStreamInstance(TEST_STREAM_NAME).serialize())).when(shardInfo).streamIdentifier();
         doReturn(Optional.empty()).when(shardInfo).streamIdentifierSerOpt();
-        retrievalFactory.createGetRecordsCache(shardInfo, mock(MetricsFactory.class));
+        retrievalFactory.createGetRecordsCache(shardInfo, streamConfig, mock(MetricsFactory.class));
         assertThat(retrievalFactory, not(nullValue()));
         verify(consumerRegistration).getOrCreateStreamConsumerArn();
     }
 
+    @Test
+    public void testRegisterNotCalledWhenConsumerArnSetInMultiStreamMode() throws Exception {
```

Comment:
Should we test that consumerArn unset in MultiStreamMode is valid also?

Reply:
Added testRegisterCalledWhenConsumerArnNotSetInMultiStreamMode

Reply:
added testRegisterCalledWhenConsumerArnNotSetInMultiStreamMode